### ISSUE TITLE
linux guest too small ("fit guest now" disabled) - sporadic

### ISSUE THREAD

**Issue by dec1** [vmware/open-vm-tools#152]
Hello,
I have VMware Workstation 12 Pro (Windows 10 Host) running a Linux OpenSuse Leap 42.2 Guest.
It regularly happens when I start the guest that it is too small and can't be resized bigger until I shutdown and restart both the guest and vmware. 
In the Linux guest I have "open-vm-tools", and "open-vm-tools-desktop" version 10.0.7-1.1-x86_64 (vendor opensuse).

I can't see any pattern to when this sizing problem occurs, and I cant seem to find a way to fix it without having to reboot  guest and restart vmware. Any ideas?

Thanks in Advance

**Comment by oliverkurth** [comment]
Thank you for reporting. I have created an internal bug for this issue, I hope you will hear back from us soon.

**Comment by awalton** [comment]
@dec1 could you attach a screenshot taken from the host so we can see what you mean? Is it the guest's resolution that is too small, or the DPI that is wrong? Attaching the logs from both the VMX running on the host and from tools also couldn't hurt: https://kb.vmware.com/selfservice/microsites/search.do?language=en_US&cmd=displayKC&externalId=1007873 for more information on how to setup tools logging in the guest.

**Comment by dec1** [comment]
Ok. When the problem arises again.

If I enable logging in the guest will the log file grow indefinitely?
Where do I find the host log?

**Comment by dec1** [comment]
![vmware_small](https://cloud.githubusercontent.com/assets/5962877/24241529/6605c3d2-0fb5-11e7-8369-1b8bc34edac7.png)
[wmware_logs.zip](https://github.com/vmware/open-vm-tools/files/864231/wmware_logs.zip)

**Comment by dec1** [comment]
Hello, above a screnshoot and the logs from the guest.
I don't know where the host logs are.

**Comment by dec1** [comment]
[vmsupport-2017-03-23-10-47.zip](https://github.com/vmware/open-vm-tools/files/864250/vmsupport-2017-03-23-10-47.zip)

**Comment by dec1** [comment]
Above also the result of "help"->"support" -> "collect support data" in my vmware workstation 12.5.4

**Comment by dec1** [comment]
![vmware_normal](https://cloud.githubusercontent.com/assets/5962877/24242417/f8e1e350-0fb7-11e7-9c8d-38f5f19b183b.png)

**Comment by dec1** [comment]
Above how it should look. The only way I can achieve this is by restarting both the guest and vmware

**Comment by microcoder** [comment]
Hello, I have a similar problem on VMWare Player 12.5.5 (Windows 10 Host) running a Kubuntu 17.04 (KDE 5.9). Under Ubuntu 17.04 this problem is not observed, only under KDE and installed package open-vm-tools, open-vm-tools-desktop v10.1.5. If i removing packages open-vm-tools (v10.1.5) and after installing vmware tools from main menu: Player->Manage->Install VMWare Tools (v10.1.6) then this problem is not observed, resolution guest OS changing, drag-n-drop working!

**Comment by oliverkurth** [comment]
@microcoder , in the non-working case (KDE), can you please check with the command:

ps awx | grep vmtoolsd

if there is a process running with the "-n vmusr" option?

**Comment by microcoder** [comment]
@oliverkurth, not there is a process on **K**ubuntu 17.04 (up-to-date):

$ ps awx | grep vmtoolsd
   981 ?        Ssl    0:00 /usr/bin/vmtoolsd
  1935 pts/1    S+     0:00 grep --color=auto vmtoolsd

[vmware-logs.tar.gz](https://github.com/vmware/open-vm-tools/files/883711/vmware-logs.tar.gz)

**Comment by oliverkurth** [comment]
Thanks. The vmusr process is needed for resolutions settings and copy/paste.

Does it work when you run:
`/usr/bin/vmware-user-suid-wrapper`
?
This should be done automatically when KDE starts, from the file /etc/xdg/autostart/vmware-user.desktop .

**Comment by microcoder** [comment]
Yes, it work:

`$ /usr/bin/vmware-user-suid-wrapper`
`vmware-user: could not open /proc/fs/vmblock/dev`

`$ ps awx | grep vmtoolsd`
`   981 ?        Ssl    0:02 /usr/bin/vmtoolsd`
`  6245 pts/1    S      0:00 /usr/bin/vmtoolsd -n vmusr`
`  6247 pts/1    S+     0:00 grep --color=auto vmtoolsd`

file /etc/xdg/autostart/vmware-user.desktop:

`[Desktop Entry]`
`Type=Application`
`Encoding=UTF-8`
`Exec=/usr/bin/vmware-user-suid-wrapper`
`Name=VMware User Agent`
`# KDE bug 190522: KDE does not autostart items with NoDisplay=true...`
`# NoDisplay=true`
`X-KDE-autostart-phase=1`

Then I restarted the system and the service running:

`$ ps awx | grep vmtoolsd`
`   877 ?        Ss     0:00 /usr/bin/vmtoolsd`
`  1944 ?        Sl     0:00 /usr/bin/vmtoolsd -n vmusr`
`  2912 pts/2    S+     0:00 grep --color=auto vmtoolsd`

Then i do power off and power on virtual machine with starting OS, service **not running**
This is working not stable. Any ideas? Thanks.

**Comment by oliverkurth** [comment]
If it only happens with Kubuntu 17.04, but not with vanilla Ubuntu 17.04, that looks like a bug in Kubuntu. I can try to reproduce this and file a bug at Kubuntu, or you can and post the bug id here.

When you said it does not happen in Ubuntu 17.04, is that with gnome?

**Comment by microcoder** [comment]
Please help file a bug at Kubuntu. Thanks.
Yes, with Unity/Gnome.

**Comment by dec1** [comment]
Were the logs and other information I provided of any help?

**Comment by oliverkurth** [comment]
I couldn't find anything useful in the logs. @dec1 can you please enable logging for the user process? Put the following lines into /etc/vmware-tools/tools.conf:

```
# Enable new "vmusr" service logging to a file.
vmusr.level = debug
vmusr.handler = file
vmusr.data = /tmp/vmware-vmusr.${USER}.log
```

I am not sure if your issue is the same as microcoder's issue - the common factor is KDE, but you say you need to restart WS. Can you please check if there is a user process of vmtoolsd running in the non-working case, as in 
https://github.com/vmware/open-vm-tools/issues/152#issuecomment-290523016 ? Will it work when you start it manually with:

`/usr/bin/vmware-user-suid-wrapper` ?

**Comment by dec1** [comment]
Hello,


Will do.

So next time I have the problem you want me to send you

/tmp/vmware-vmusr.${USER}.log ?

**Comment by oliverkurth** [comment]
Yes, please, in addition to checking if the vmtoolsd user process is running.

Please also attach the vmware.log file. This should be in the same directory where the data of the VM are stored.

Thanks!

**Comment by dec1** [comment]
Hello Oliver.

> ps awx | grep vmtoolsd
 1219 ?        Sl     0:00 /usr/bin/vmtoolsd
 2552 pts/0    S+     0:00 grep --color=auto vmtoolsd

>  /usr/bin/vmware-user-suid-wrapper

# guest resized properly ..thanks now I dont need to restart (sometimes multiple times) each time!

> ps awx | grep vmtoolsd

 1219 ?        Sl     0:00 /usr/bin/vmtoolsd
 2656 pts/0    Sl     0:00 /usr/bin/vmtoolsd -n vmusr
 2932 pts/0    S+     0:00 grep --color=auto vmtoolsd

**Comment by dec1** [comment]
[vmware-vmusr.declan.zip](https://github.com/vmware/open-vm-tools/files/918785/vmware-vmusr.declan.zip)

The logs as requested

**Comment by dec1** [comment]
[vmware.zip](https://github.com/vmware/open-vm-tools/files/918787/vmware.zip)

**Comment by microcoder** [comment]
Hello, Oliver!

Did you registered a bug at Kubuntu, or not? Thanks.

**Comment by oliverkurth** [comment]
Sorry @microcoder , I wasn't able to create a setup yet to reproduce. I was out last week, and when I tried a fresh installation for 17.04-beta1 didn't work for me. I need to try again.

**Comment by oliverkurth** [comment]
@microcoder , I was able to reproduce the issue, and reported a bug on Launchpad: https://bugs.launchpad.net/ubuntu/+source/open-vm-tools/+bug/1682647 . Feel free to comment there.

**Comment by oliverkurth** [comment]
@dec, thank you for the log files. We are going to analyze them.
